test/dynamo/test_higher_order_ops.py

```diff
@@ -0,0 +1,351 @@
+# Owner(s): ["module: dynamo"]
+import re
+import unittest
+
+import torch
+
+import torch._dynamo.test_case
+from torch._dynamo.utils import counters
+from torch._ops import wrap
+
+
+class MockBackend:
+    def __init__(self):
+        self.graphs = []
+
+    def __call__(self, gm: torch.fx.GraphModule, example_inputs):
+        self.graphs.append(gm)
+        return gm.forward
+
+
+global_var = torch.randn(3)
+global_num = 3.14
+
+
+class TestHigherOrderOps(torch._dynamo.test_case.TestCase):
+    def test_no_freevars(self):
+        mock = MockBackend()
+
+        def f(x):
+            return wrap(lambda x: torch.sin(x), x)
+
+        x = torch.randn(3)
+        expected = f(x)
+        result = torch.compile(f, backend=mock)(x)
+
+        self.assertEqual(result, expected)
+        self.assertEqual(len(mock.graphs), 1)
+        self.assertIsNotNone(re.search(r"wrap\(\w+, \w+\);", mock.graphs[0].code))
+
+    def test_capture_untracked_global(self):
+        counters.clear()
+        mock = MockBackend()
+
+        def f(x):
+            return wrap(lambda x: x + global_var, x)
+
+        x = torch.randn(3)
+        expected = f(x)
+        result = torch.compile(f, backend=mock)(x)
```

Comment:
IMO prefer testing `torch._dynamo.optimize("eager")` - that's the frontend. This invokes aot_autograd and inductor, which is a fuller test. Keep unit tests to a unit (graph capture, in this case). 

Both are still worth having, but maybe lower down, or alongside this.